# 2.0.1 Números Factoriales


In [1]:
%matplotlib notebook
import numpy as np
import matplotlib.pyplot as plt

**1. Escriba una función que calcule el factorial de n, con n en los números naturales. Calcule los primeros 20 números factoriales**

In [12]:
def getFactorial(n):
    return np.math.factorial(n)

for i in range(20):
    print(getFactorial(i))
    

1
1
2
6
24
120
720
5040
40320
362880
3628800
39916800
479001600
6227020800
87178291200
1307674368000
20922789888000
355687428096000
6402373705728000
121645100408832000


**2. Escriba una función que calcule las variaciones sin repetición de n elemenos tomados de
r en r:**

**a) Calcule de cuantas maneras puedo ubicar 6 carros en 3 estacionamientos. Ans: 120**



In [22]:
def getVariation(n, r):
    return np.math.factorial(n)/np.math.factorial(n-r)

print(getVariation(6, 3))

120.0
